### PR TITLE
Return null by default for byte[] getters

### PR DESCRIPTION
**Pull request by alzimmermsft**
Changes the default behavior in `ArrayType` to return `null` instead of `new byte[0]` when the `defaultValueExpression` is null. To provide a way for backwards compatibility a new configuration `default-byte-array-returns-empty-array` has been added to `JavaSettings`. Additionally, the previous behavior has a minor improvement by reusing a static singleton for the empty byte array instead of creating a new instance on each get call.

### CODE REVIEW

File: extension-base/src/main/java/com/azure/autorest/extension/base/plugin/JavaSettings.java [hunk 1]
```diff
@@ -169,7 +169,8 @@ public static JavaSettings getInstance() {
                 // were generated with required = true set in JsonProperty annotation
                 getBooleanValue(host, "disable-required-property-annotation", false),
                 getBooleanValue(host, "enable-page-size", false),
-                getBooleanValue(host, "use-key-credential", false)
+                getBooleanValue(host, "use-key-credential", false),
+                getBooleanValue(host, "default-byte-array-returns-empty-array", false)
```
Comment (srnagar) [inline extension-base/src/main/java/com/azure/autorest/extension/base/plugin/JavaSettings.java:173]: `null-byte-array-maps-to-empty-array`

File: javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java [hunk 2]
```diff
@@ -132,6 +132,15 @@ public final void write(ClientModel model, JavaFile javaFile) {
                 classBlock.privateStaticFinalVariable("Pattern KEY_ESCAPER = Pattern.compile(\"\\\\.\");");
             }
 
+            // If code is being generated with the behavior to return an empty byte array when the default value
+            // expression is null and the model has any array types that will need conversion within getter methods
+            // generate a static byte[] that will be returned instead of creating a new instance each get.
+            if (settings.isDefaultByteArrayReturnsEmptyArray()
+                && model.getProperties().stream().anyMatch(property -> property.getClientType() instanceof ArrayType
+                    && property.getWireType() != property.getClientType())) {
+                classBlock.privateStaticFinalVariable("byte[] EMPTY_BYTE_ARRAY = new byte[0]");
```
Comment (srnagar) [inline javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java:141]: Do we not have a constant in azure-core for this?

**Review by srnagar — APPROVED** [review]

File: javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java (hunk 2 above)
Comment (alzimmermsft) [reply javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java:141]: Nothing that is public

**Comment by weidongxu-microsoft** [comment]
null array could trigger some spotbug error

```java
    public byte[] data() {
        if (this.data == null) {
            return null;
        }
        return this.data.decodedBytes();
    }
```

see https://github.com/Azure/autorest.java/pull/2334

**Pull request merged**